### PR TITLE
[issue 86] Create a demo that uses other widget libraries in a dashboard

### PR DESCRIPTION
**Pull request by nitind**
Add a demo notebook featuring a bokeh plot.

### CODE REVIEW

**Comment by nitind** [comment]
For issue 86

File: etc/notebooks/bokeh_demo/bokeh_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+       "width": 12
+      }
+     }
+    }
+   },
+   "source": [
+    "Bokeh Dashboard Demo\n",
+    "---------------------\n",
+    "\n",
+    "This demo shows off an interactive visualization using [Bokeh](http://bokeh.pydata.org) for plotting, and Ipython interactors for widgets. It is nearly identical to the [demo notebook](https://github.com/bokeh/bokeh/blob/master/examples/plotting/notebook/interact_basic.ipynb) found in Bokeh's own repository."
```
Comment (parente) [inline etc/notebooks/bokeh_demo/bokeh_demo.ipynb]: Ipython -> IPython

File: etc/notebooks/bokeh_demo/bokeh_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "source": [
+    "The demo runs entirely inside the Ipython notebook, with no Bokeh server required.\n",
```
Comment (parente) [inline etc/notebooks/bokeh_demo/bokeh_demo.ipynb]: Ipython -> Jupyter

File: etc/notebooks/bokeh_demo/bokeh_demo.ipynb (hunk 1 above)
Comment (parente) [inline etc/notebooks/bokeh_demo/bokeh_demo.ipynb]: Also, call out that a user can switch to dashboard layout and view modes to interact with these widgets.

**Comment by nitind** [comment]
Updated with case fixes and information about the **dashboard layout** and **dashboard view** views.

**Comment by parente** [comment]
When I run the conda install, it upgrades a ton of stuff. After that, when I run the cell that does the imports, I get a long stack trace that ends with:

```
OSError: zipimport: can not open file /opt/conda/lib/python3.4/site-packages/setuptools-18.1-py3.4.egg
```

Any way to prevent conda from upgrading the world? Or any idea what you did differently?

**Comment by parente** [comment]
Switching to pip install fixes the problem because pip doesn't go upgrading unnecessary dependencies. conda is upgrading itself, which pulls in a new version of setuptools (0.18.4), which apparently bokeh doesn't like and tries to load 0.18.1 explicitly instead. Maybe because the kernel was started with that earlier dep installed?

Besides that problem, the cells in the dashboard layout need to be sized to contain the widgets at their default size. By default, I see them overflowing.

And one question: are the two text cells talking about installing bokeh and switching modes intended to be in the dashboard view or accidentally left there?

**Comment by nitind** [comment]
I haven't seen the overflowing issue when running locally, in any browser, but I did see those cells were below the fold after you pointed them out. I fiddled with the layout slightly, but I'm not sure how to attack that one otherwise.

**Pull request closed without merging**